library/src/test/java/com/yayandroid/locationmanager/mocks/MockSimpleTask.java

```diff
@@ -0,0 +1,204 @@
+        }
+    }
+
+    @Nullable
+    public Exception getException() {
+        return this.error;
+    }
+
+    @NonNull
+    public Task<TResult> addOnSuccessListener(@NonNull OnSuccessListener<? super TResult> var1) {
+        if (isSuccessful()) {
+            var1.onSuccess(result);
+        }
+
+        return this;
+    }
+
+    @NonNull
+    public Task<TResult> addOnSuccessListener(@NonNull Executor var1, @NonNull OnSuccessListener<? super TResult> var2) {
+        if (isSuccessful()) {
+            var2.onSuccess(result);
+        }
+
+        return this;
+    }
+
+    @NonNull
+    public Task<TResult> addOnSuccessListener(@NonNull Activity var1, @NonNull OnSuccessListener<? super TResult> var2) {
+        if (isSuccessful()) {
+            var2.onSuccess(result);
+        }
+
+        return this;
+    }
+
+    @NonNull
+    public Task<TResult> addOnFailureListener(@NonNull OnFailureListener var1) {
+        if (error != null && !isSuccessful()) {
+            var1.onFailure(error);
+        }
+
+        return this;
+    }
+
+    @NonNull
+    public Task<TResult> addOnFailureListener(@NonNull Executor var1, @NonNull OnFailureListener var2) {
+        if (error != null && !isSuccessful()) {
+            var2.onFailure(error);
+        }
+
+        return this;
+    }
+
+    @NonNull
+    public Task<TResult> addOnFailureListener(@NonNull Activity var1, @NonNull OnFailureListener var2) {
+        if (error != null && !isSuccessful()) {
+            var2.onFailure(error);
+        }
+
+        return this;
+    }
+
+    @NonNull
+    public Task<TResult> addOnCompleteListener(@NonNull OnCompleteListener<TResult> var1) {
+        if (isComplete()) {
+            var1.onComplete(this);
+        }
+
+        return this;
+    }
+
+    @NonNull
+    public Task<TResult> addOnCompleteListener(@NonNull Executor var1, @NonNull OnCompleteListener<TResult> var2) {
+        if (isComplete()) {
+            var2.onComplete(this);
+        }
+
+        return this;
+    }
+
+    @NonNull
+    public Task<TResult> addOnCompleteListener(@NonNull Activity var1, @NonNull OnCompleteListener<TResult> var2) {
+        if (isComplete()) {
+            var2.onComplete(this);
+        }
+
+        return this;
+    }
+
+    @NonNull
+    public Task<TResult> addOnCanceledListener(@NonNull OnCanceledListener var1) {
+        if (isCanceled()) {
+            var1.onCanceled();
+        }
+
+        return this;
+    }
+
+    @NonNull
+    public Task<TResult> addOnCanceledListener(@NonNull Executor var1, @NonNull OnCanceledListener var2) {
+        if (isCanceled()) {
+            var2.onCanceled();
+        }
+
+        return this;
+    }
+
+    @NonNull
+    public Task<TResult> addOnCanceledListener(@NonNull Activity var1, @NonNull OnCanceledListener var2) {
+        if (isCanceled()) {
+            var2.onCanceled();
+        }
+
+        return this;
+    }
+
+    public void setResult(@Nullable TResult var1) {
+        this.result = var1;
```

Comment:
Can you call `success` and `completed` here?
So that you don't need to override `isSuccessful` and `isComplete`

Reply:
No, listener not stored in class. MockSimpleTask updated, unused methods removed.